src/kOS.Safe/Compilation/Opcode.cs

```diff
@@ -1074,12 +1062,15 @@ public override void Execute(ICpu cpu)
                 Type t = value.GetType();
                 MethodInfo negateMe = t.GetMethod("op_UnaryNegation", BindingFlags.FlattenHierarchy |BindingFlags.Static | BindingFlags.Public); // C#'s alternate name for '-' operator
                 if (negateMe != null)
-                    result = negateMe.Invoke(null, new[]{value}); // value is an arg, not the 'this'.  (Method is static.)
+                {
+                    object result = negateMe.Invoke(null, new[]{value});
+                    scalarValue = ScalarValue.Create(result);
```

Comment:
This is a good place to use `Structure.FromPrimitive` instead.  It's basically the same thing as ScalarValue.Create but with the caveat that if it can't convert it to a scalar, it doesn't throw an exception it just returns it as-is.